src/AbstractMigrator.php

```diff
@@ -0,0 +1,395 @@
+        } catch (RootException $exception) {
+            throw $this->_createCouldNotMigrateException(
+                $this->__('Failed to migrate up'),
+                null,
+                $exception,
+                $upVersion
+            );
+        }
+    }
+
+    /**
+     * {@inheritdoc}
+     *
+     * Overridden to wrap around the parent method and re-throw any exceptions as specific exceptions.
+     *
+     * @since [*next-version*]
+     */
+    protected function _down($upVersion = null, $force = false)
+    {
+        try {
+            parent::down($upVersion, $force);
+        } catch (RootException $exception) {
+            throw $this->_createCouldNotMigrateException(
+                $this->__('Failed to migrate down'),
+                null,
+                $exception,
+                $upVersion
+            );
+        }
+    }
+
+    /**
+     * Resets the database to the base version, and then optionally up to a specific version.
+     *
+     * Almost identical to {@link ByjgMigration::reset()}, except that it catches and throws
+     * specific exceptions and does not attempt to perform "up" migration after resetting.
+     *
+     * @since [*next-version*]
+     */
+    protected function _reset()
+    {
+        try {
+            try {
+                $this->getCurrentVersion();
+                $this->down(0);
+            } catch (DatabaseNotVersionedException $versionedException) {
+                // Do nothing
+            }
+
+            $db = $this->getDbCommand();
+            $db->createVersion();
+            $db->setVersion(0, 'complete');
+        } catch (RootException $exception) {
+            throw $this->_createCouldNotMigrateException(
+                $this->__('Failed to reset'),
+                null,
+                $exception,
+                0
+            );
+        }
+    }
+
+    /**
+     * Method for execute the migration.
+     *
+     * Overridden to:
+     * * pass different arguments to the callback progress function
+     * * allow preparation of SQL query strings before execution
+     * * contain a portion of the fix for the SQL file execution bug See {@link getMigrationSqlQuery()}.
+     * * throw specific exceptions
+     *
+     * @param int  $upVersion
+     * @param int  $increment Can accept 1 for UP or -1 for down
+     * @param bool $force
+     *
+     * @throws DatabaseIsIncompleteException
+     */
+    protected function migrate($upVersion, $increment, $force)
+    {
+        $versionInfo    = $this->getCurrentVersion();
+        $currentVersion = intval($versionInfo['version']);
+
+        if (strpos($versionInfo['status'], 'partial') !== false && !$force) {
+            throw new DatabaseIsIncompleteException(
+                'Database was not fully updated - use the "force" argument to ignore this error'
+            );
+        }
+
+        while (
+            $this->canContinue($currentVersion, $upVersion, $increment)
+            &&
```

Comment:
Does the `&&` really need a line of its own?

Reply:
Probably not. I'll invoke the CS fixer and see if it fixes it and report back.

Reply:
Kemm inti invowker